folio-spring-i18n/src/main/java/org/folio/spring/service/TranslationService.java

```diff
@@ -0,0 +1,347 @@
+            return new TranslationMap(
+              locale,
+              languageMap.get(locale.getCountry().toLowerCase()),
+              baseMap
+            );
+          }
+
+          return baseMap;
+        }
+
+        return base;
+      }
+    );
+  }
+
+  /**
+   * Get the best {@link TranslationMap TranslationMap} associated with the current locale.
+   *
+   * @param locale the locale to find a TranslationMap for
+   * @return the best translation available for the current locale, including the server's default
+   */
+  public TranslationMap getTranslation(Locale locale) {
+    return getTranslation(locale, getDefaultTranslation().withLocale(locale));
+  }
+
+  /**
+   * Find a TranslationMap for the default locale -- used to initialize for {@code getDefaultTranslations}.
+   *
+   * @return the best applicable translation
+   * @throws IllegalStateException if no translation can be found
+   */
+  protected TranslationMap resolveDefaultLocale() {
+    TranslationMap foundDefault = getTranslation(
+      configuration.getDefaultLocale(),
+      null
+    );
+    if (foundDefault == null) {
+      foundDefault = getTranslation(Locale.ENGLISH, null);
+    }
+    if (foundDefault == null) {
+      throw new IllegalStateException(
+        String.format(
+          "No translations are sufficient for the server's default locale (%s) nor %s",
+          configuration.getDefaultLocale(),
+          Locale.ENGLISH
+        )
+      );
+    }
+    return foundDefault;
+  }
+
+  /**
+   * Get the translation map for the default locale (or en-us, if that is not possible).
+   *
+   * @return the default locale's translation map
+   */
+  public TranslationMap getDefaultTranslation() {
+    // computeIfAbsent does not work due to the resolver potentially filling multiple keys
+    if (
+      !this.localeTranslations.containsKey(configuration.getDefaultLocale())
+    ) {
+      this.localeTranslations.put(
+        configuration.getDefaultLocale(),
+        this.resolveDefaultLocale()
+      );
+    }
+    return this.localeTranslations.get(configuration.getDefaultLocale());
+  }
+
+  /**
+   * Get all of the locales for the current request.  This will respect the Accept-Language header
+   * and quality values, as applicable, with highest quality first.
+   *
+   * @return a List of all Locale objects.  Invalid and wildcard Locales will likely return a Locale
+   *   with empty string/fields or, in some cases, the server's current Locale.
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language">MDN docs</a>
+   */
+  @SuppressWarnings("java:S1166")
+  public List<Locale> getCurrentLocales() {
+    try {
+      return Collections.list(
+        (
+          (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()
+        ).getRequest()
+          .getLocales()
+      );
+    } catch (IllegalStateException e) {
```

Comment:
This looks like it's silently swallowing the exception, so I'd at least add a comment explaining why that's okay